src/scanpy/preprocessing/_highly_variable_genes.py

```diff
@@ -538,6 +572,7 @@ def highly_variable_genes(  # noqa: PLR0913
     subset: bool = False,
     inplace: bool = True,
     batch_key: str | None = None,
+    filter_unexpressed_genes: bool = False,
```

Comment:
This parameter is still unused here, passing it to `highly_variable_genes` does nothing.

Can you please either remove it again or:

1. set the default to `None`, since it’s derived from `batch_key`.
2. explain in the parameter docs and a comment on the `else` branch where `_highly_variable_genes_batched` is called, that the parameter is always `True` if `batch_key` is `False`
3. raise a warning if an incompatible combination is chosen (i.e. `batch_key="..."` and `filter_unexpressed_genes=False`)
4. actually pass it on to `_highly_variable_genes_single_batch` and add a test that checks that `highly_variable_genes(..., batch_key=None, filter_unexpressed_genes=True)` behaves differently from `highly_variable_genes(..., batch_key=None, filter_unexpressed_genes=False)`